spec/models/visitor_spec.rb

```diff
@@ -21,4 +21,10 @@
   it "has many unsynced assignments" do
     expect(subject).to have_many(:unsynced_splits)
   end
+
+  it "doesn't include assignments to finished splits" do
+    dead_split = FactoryBot.create(:split, finished_at: Time.zone.now)
+    dead_assignment = FactoryBot.create(:assignment, split: dead_split, visitor: subject)
+    expect(subject.reload.assignments).not_to include(dead_assignment)
+  end
```

Comment:
This is an interesting, we've always included all assignments ever, even for finished splits. This was basically a bug. Now it works correctly.

Reply:
(this should improve performance for longstanding customers and employees a bit)

Reply:
Nice! 👍